sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -188,7 +188,8 @@ case class Mode(
 
   assert(orderingFilled || (!orderingFilled && reverseOpt.isEmpty))
 
-  override def withOrderingWithinGroup(orderingWithinGroup: Seq[SortOrder]): AggregateFunction = {
+  override def withOrderingWithinGroup(
+      orderingWithinGroup: Seq[SortOrder], u: UnresolvedFunction): AggregateFunction = {
```

Comment:
It's pretty weird to have `UnresolvedFunction` parameter in this API `withOrderingWithinGroup`.

I'm not quite convinced by this refactoring. `ListAgg` is probably the only special case and I think it's fine to have a dedicated check for this special case in `CheckAnalysis`. We can revisit it later when we do need to add more special cases.

Reply:
Okay, let's keep this PR pending for now and review it later.